codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaderDateFormatter.java

```diff
@@ -0,0 +1,421 @@
+            } else if ("Aug".equalsIgnoreCase(tokenString)) {
+                month = Calendar.AUGUST;
+            } else if ("Sep".equalsIgnoreCase(tokenString)) {
+                month = Calendar.SEPTEMBER;
+            } else if ("Oct".equalsIgnoreCase(tokenString)) {
+                month = Calendar.OCTOBER;
+            } else if ("Nov".equalsIgnoreCase(tokenString)) {
+                month = Calendar.NOVEMBER;
+            } else if ("Dec".equalsIgnoreCase(tokenString)) {
+                month = Calendar.DECEMBER;
+            }
+        }
+
+        return month != -1;
+    }
+
+    private boolean tryParseYear(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        if (len != 2 && len != 4) {
+            return false;
+        }
+
+        int localYear = 0;
+        for (int i = tokenStart; i < tokenEnd; i++) {
+            char c = txt.charAt(i);
+            if (isDigit(c)) {
+                localYear = localYear * 10 + getNumericalValue(c);
+            } else {
+                // invalid
+                return false;
+            }
+        }
+
+        if (localYear > 0) {
+            year = localYear;
+            return true;
+        }
+
+        return false;
+    }
+
+    private void parseToken(CharSequence txt, int tokenStart, int tokenEnd) {
+        if (!timeFound) {
+            timeFound = tryParseTime(txt, tokenStart, tokenEnd);
+            if (timeFound) {
+                return;
+            }
+        }
+
+        if (!dayOfMonthFound) {
+            dayOfMonthFound = tryParseDayOfMonth(txt, tokenStart, tokenEnd);
+            if (dayOfMonthFound) {
+                return;
+            }
+        }
+
+        if (!monthFound) {
+            monthFound = tryParseMonth(txt, tokenStart, tokenEnd);
+            if (monthFound) {
+                return;
+            }
+        }
+
+        if (!yearFound) {
+            yearFound = tryParseYear(txt, tokenStart, tokenEnd);
+        }
+    }
+
+    private Date parse0(CharSequence txt, int start, int end) {
+        int tokenStart = -1;
+
+        for (int i = start; i < end; i++) {
+            char c = txt.charAt(i);
+
+            if (isDelim(c)) {
+                if (tokenStart != -1) {
+                    // terminate token
+                    parseToken(txt, tokenStart, i);
+                    tokenStart = -1;
+                }
+            } else {
+                if (tokenStart == -1) {
+                    // start new token
+                    tokenStart = i;
+                }
+            }
+        }
+
+        if (tokenStart != -1) {
+            // terminate trailing token
+            parseToken(txt, tokenStart, txt.length());
+        }
+
+        if (!timeFound || !dayOfMonthFound || !monthFound || !yearFound) {
+            // missing field
+            return null;
+        }
+
+         if (dayOfMonth < 1 || dayOfMonth > 31 || hours > 23 || minutes > 59 || seconds > 59) {
```

Comment:
I would consider at least combining the 2 conditionals. Currently it looks like there are duplicate checks between the 2.